src/pooledembeddings.jl

```diff
@@ -0,0 +1,87 @@
+export PooledEmbeddings
+
+using ChainRulesCore: NoTangent
+using Polynomials4ML: PooledSparseProduct, AbstractPoly4MLBasis
+
+import ChainRulesCore: rrule
+import Base.Cartesian: @nexprs
+
+struct PooledEmbeddings{NB, TB <: Tuple} <: AbstractPoly4MLBasis
+   embeddings::TB
+   pooling::PooledSparseProduct{NB}
+   @reqfields()
+end
+
+function PooledEmbeddings(embeddings, pooling)
+   return PooledEmbeddings(embeddings, pooling, _make_reqfields()...)
+end
+
+Base.length(basis::PooledEmbeddings) = length(basis.pooling)
+
+function _write_code_Bi_tup(NB)
```

Comment:
this won't work if you have a single basis because `(B_1)` doesn't define a tuple. Rewrite as follows: 
```julia
_write_code_Bi_tup(NB) = Meta.parse( "(" * prod("B_$i, " for i = 1:NB) * ")" )
```